packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/diagram/ViewToolSectionsProvider.java

```diff
@@ -68,44 +105,188 @@ public List<ToolSection> handle(Object targetElement, Object diagramElement, Obj
+            // @formatter:off
+            String toolId = this.idProvider.apply(nodeTool).toString();
+            var tool = SingleClickOnDiagramElementTool.newSingleClickOnDiagramElementTool(toolId)
+                    .label(nodeTool.getName())
+                    .imageURL(ViewToolImageProvider.NODE_CREATION_TOOL_ICON)
+                    //Empty handler
+                    .handler(variableManager -> {
+                        return null;
+                    })
+                    .targetDescriptions(List.of())
+                    .appliesToDiagramRoot(true)
+                    .build();
+            // @formatter:on
+            diagramTools.add(tool);
+        }
+        return diagramTools;
     }
 
-    private List<ToolSection> getNodeToolSections(DiagramDescription diagramDescription, NodeDescription nodeDescription) {
+    private List<ToolSection> getNodePalette(DiagramDescription diagramDescription, NodeDescription nodeDescription) {
         String sourceElementId = this.getSourceElementId(nodeDescription.getId());
+        // Palette for the diagram itself
         String nodePaletteId = "siriusComponents://nodePalette?nodeId=" + sourceElementId;
-        return this.findToolSectionById(diagramDescription, nodePaletteId).stream().toList();
+        var allToolSections = new ArrayList<ToolSection>();
+        var optionalNodeDescription = this.viewRepresentationDescriptionSearchService.findViewNodeDescriptionById(sourceElementId);
+
+        if (optionalNodeDescription.isPresent()) {
+            org.eclipse.sirius.components.view.NodeDescription viewNodeDescription = optionalNodeDescription.get();
+            // @formatter:off
+            var nodePalette = ToolSection.newToolSection(nodePaletteId)
+                    .label(viewNodeDescription.getName())
+                    .tools(this.createNodePaletteTools(diagramDescription, viewNodeDescription, nodeDescription))
+                    .imageURL("")
+                    .build();
+            // @formatter:on
+
+            allToolSections.add(nodePalette);
+
+        }
+        return allToolSections;
+    }
+
+    private List<ITool> createNodePaletteTools(DiagramDescription diagramDescription, org.eclipse.sirius.components.view.NodeDescription viewNodeDescription, NodeDescription nodeDescription) {
+        List<ITool> tools = new ArrayList<>();
+
+        for (NodeTool nodeTool : new ToolFinder().findNodeTools(viewNodeDescription)) {
+            // @formatter:off
+            String toolId = this.idProvider.apply(nodeTool).toString();
+            var tool = SingleClickOnDiagramElementTool.newSingleClickOnDiagramElementTool(toolId)
+                    .label(nodeTool.getName())
+                    .imageURL(ViewToolImageProvider.NODE_CREATION_TOOL_ICON)
+                    //Empty handler
+                    .handler(variableManager -> {
+                        return null;
```

Comment:
Add a failure handler with the unexpected error message.
Log an error in the console as well